### PR TITLE
Fix most of the warnings in hash_engine

### PR DESCRIPTION
**Pull request by TimDiekmann**

### CODE REVIEW

File: packages/engine/src/simulation/package/state/packages/behavior_execution/mod.rs [hunk 1]
```diff
@@ -34,7 +34,7 @@ pub type BehaviorIdInnerDataType = u16;
 pub type BehaviorIndexInnerDataType = f64;
 
 pub struct Creator {
-    experiment_config: Option<Arc<ExperimentConfig>>,
+    _experiment_config: Option<Arc<ExperimentConfig>>,
```
Comment (Alfred-Mountfield) [inline packages/engine/src/simulation/package/state/packages/behavior_execution/mod.rs:37]: This one actually feels weird to me, I think we can just delete these. Packages can decide what they want to keep in the Creator so no need to keep it for consistency

File: packages/engine/src/simulation/package/state/packages/behavior_execution/mod.rs [hunk 2]
```diff
@@ -108,7 +108,7 @@ impl PackageCreator for Creator {
             .index_of(behavior_index_col.value())?;
 
         Ok(Box::new(BehaviorExecution {
-            behavior_map: Arc::clone(self.get_behavior_map()?),
+            _behavior_map: Arc::clone(self.get_behavior_map()?),
```
Comment (Alfred-Mountfield) [inline packages/engine/src/simulation/package/state/packages/behavior_execution/mod.rs:111]: Ditto for these ,if they're unused, no need to store them. Unless there's something to do with them being dropped early or whatever, in which case definitely needs a comment. My guess is that they're unneeded in the struct

File: packages/engine/src/simulation/package/state/packages/behavior_execution/reset_index_col.rs [hunk 3]
```diff
@@ -1,15 +1,9 @@
...
-    datastore::{
-        arrow::batch_conversion::{new_buffer, new_offsets_buffer},
-        batch::iterators,
-    },
-    simulation::package::state::packages::behavior_execution::config::BehaviorId,
-};
+use crate::datastore::arrow::batch_conversion::new_buffer;
 
-pub fn reset_index_col(state: &ExState, behavior_index_col_index: usize) -> Result<StateColumn> {
+pub fn reset_index_col(_state: &ExState, behavior_index_col_index: usize) -> Result<StateColumn> {
```
Comment (Alfred-Mountfield) [inline packages/engine/src/simulation/package/state/packages/behavior_execution/reset_index_col.rs:6]: Doesn't seem to be a trait, so again lets delete
```suggestion
pub fn reset_index_col(behavior_index_col_index: usize) -> Result<StateColumn> {
```

@litvand, do we think we're missing something here? Any idea why state was originally included

Comment (Alfred-Mountfield) [reply packages/engine/src/simulation/package/state/packages/behavior_execution/reset_index_col.rs:6]: I am unable to suggest a change on this part of the code but a related change will need to be made manually:

```
    fn reset_behavior_index_col(&mut self, state: &mut ExState) -> Result<()> {
        let behavior_index_col = reset_index_col(state, self.behavior_index_col_index)?;
```
|
V
```
    fn reset_behavior_index_col(&mut self) -> Result<()> {
        let behavior_index_col = reset_index_col(self.behavior_index_col_index)?;
```

File: packages/engine/src/simulation/package/state/packages/behavior_execution/reset_index_col.rs [hunk 4]
```diff
@@ -25,7 +19,7 @@ impl IntoArrowChange for ResetIndexCol {
 
         // new_buffer delegates to a method that zeroes the memory so we don't need to initialize
         let mut data = new_buffer::<BehaviorIndexInnerDataType>(num_agents);
-        let mut_data = data.typed_data_mut::<BehaviorIndexInnerDataType>();
+        let _mut_data = data.typed_data_mut::<BehaviorIndexInnerDataType>();
```
Comment (Alfred-Mountfield) [inline packages/engine/src/simulation/package/state/packages/behavior_execution/reset_index_col.rs:22]: Pretty sure we can delete this as we don't initialize the memory as per the comment above
```suggestion
```

File: packages/engine/src/worker/runner/javascript/mini_v8/function.rs [hunk 5]
```diff
@@ -104,7 +104,7 @@ fn manually_drop_arg_descs(arg_descs: Vec<ValueDesc>) {
     // Ownership of the arguments' value pointers was taken by C++. They've already been properly
     // dropped:
     for desc in arg_descs.into_iter() {
-        ManuallyDrop::new(desc);
+        mem::forget(desc);
```
Comment (Alfred-Mountfield) [inline packages/engine/src/worker/runner/javascript/mini_v8/function.rs:107]: Can we double check this is correct. **If** the "value pointer" is taken by C++ (I have not checked if this is true or what exactly is taken), do we need to worry about the rest of the `ValueDesc` struct? If it's still cleared up with this then no worries

File: packages/engine/src/worker/runner/javascript/mod.rs [hunk 6]
```diff
@@ -227,11 +227,11 @@ impl<'m> Embedded<'m> {
         Ok(Self {
             hash_stdlib,
             hash_util,
-            batches_prototype,
-            experiment_ctx_prototype,
-            sim_init_ctx_prototype,
-            gen_ctx,
-            gen_state,
+            _batches_prototype: batches_prototype,
```
Comment (Alfred-Mountfield) [inline packages/engine/src/worker/runner/javascript/mod.rs:230]: @litvand Do we need all of these still?

File: packages/engine/src/worker/runner/javascript/mini_v8/function.rs (hunk 5 above)
Comment (TimDiekmann) [reply packages/engine/src/worker/runner/javascript/mini_v8/function.rs:107]: In any case, `mem::forget` is defined like this:
```rust
pub const fn forget<T>(t: T) {
    let _ = ManuallyDrop::new(t);
}
```

Comment (Alfred-Mountfield) [reply packages/engine/src/worker/runner/javascript/mini_v8/function.rs:107]: Yeah well at least this isn't a regression, just making sure we aren't covering a potential bug

File: packages/engine/src/worker/runner/javascript/mod.rs (hunk 6 above)
Comment (litvand) [reply packages/engine/src/worker/runner/javascript/mod.rs:230]: No, I didn't notice that they aren't used outside of this function, where they are available as local variables anyways.

**Pull request merged**